run/sigterm-handler/main.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample sigterm-handler shows how to gracefully shut down in response to a SIGTERM signal.
+package main
+
+// [START cloudrun_sigterm_handler]
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func main() {
+	// Determine port for HTTP service.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("defaulting to port %s", port)
+	}
+
+	srv := &http.Server{
+		Addr:    ":" + port,
+		Handler: http.HandlerFunc(handler),
+	}
+
+	// Create channel to listen for signals.
+	signalChan := make(chan os.Signal, 1)
+	// SIGINT handles Ctrl+C locally.
+	// SIGTERM handles Cloud Run termination signal.
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Start HTTP server.
+	go func() {
+		log.Printf("listening on port %s", port)
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+
+	// Receive output from signalChan.
+	sig := <-signalChan
+	log.Printf("%s signal caught", sig)
+	log.Print("server stopped")
+
+	// Timeout if waiting for connections to return idle.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
+	// Add extra handling here to close any DB connections, redis, or flush logs.
+
+	// Release resources.
+	defer cancel()
```

Comment:
```suggestion
	// Timeout if waiting for connections to return idle.
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	
	// Add extra handling here to close any DB connections, Redis, or flush logs.

```

(Missing a verb on Redis for parallel construction?)